src/Adapter/Carrier/Repository/CarrierRepository.php

```diff
@@ -102,12 +102,36 @@ public function updateInNewVersion(CarrierId $carrierId, Carrier $carrier): Carr
         if ($carrier->delay) {
             $newCarrier->delay = $carrier->delay;
         }
+        if ($carrier->max_width) {
+            $newCarrier->max_width = $carrier->max_width;
+        }
+        if ($carrier->max_height) {
+            $newCarrier->max_height = $carrier->max_height;
+        }
+        if ($carrier->max_depth) {
+            $newCarrier->max_depth = $carrier->max_depth;
+        }
+        if ($carrier->max_weight) {
+            $newCarrier->max_weight = $carrier->max_weight;
+        }
         $newCarrier->deleted = false; // just to be sure...
 
         // Copy all others information like ranges, shops associated, ...
         $newCarrier->copyCarrierData($carrierId->getValue());
         $this->updateObjectModel($newCarrier, CannotUpdateCarrierException::class);
 
-        return new CarrierId($newCarrier->id);
+        $newCarrier->setGroups($this->getGroupAccess($oldCarrier));
+
+        return $newCarrier;
+    }
+
+    public function getGroupAccess(Carrier $carrier): array
+    {
+        return array_map(fn ($group) => (int) $group['id_group'], $carrier->getGroups());
+    }
+
+    public function setGroupsAccess(Carrier $carrier, array $groups): void
+    {
+        $carrier->setGroups($groups);
```

Comment:
I'm not sure these methods have much usage as public methods, if they are used internally they could be protected but besides that they are nearly getters/setters on another type of object The repository's responsibility is to fetch data not to be used as an externalisation of getters/setters for an object

The setter method especially can always be replaced by `$carrier->setGroups($groups);` so it brings no extra values, the getter method could moved into the Carrier object model class instead in a new public methods `getAssociatedGroupIds`